src/renderer/components/dock/logs/__test__/download-logs.test.tsx

```diff
@@ -0,0 +1,186 @@
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { DiContainer } from "@ogre-tools/injectable";
+import type { RenderResult } from "@testing-library/react";
+import { act, waitFor } from "@testing-library/react";
+import { SearchStore } from "../../../../search-store/search-store";
+import openSaveFileDialogInjectable from "../../../../utils/save-file.injectable";
+import type { ApplicationBuilder } from "../../../test-utils/get-application-builder";
+import { getApplicationBuilder } from "../../../test-utils/get-application-builder";
+import type { TabId } from "../../dock/store";
+import dockStoreInjectable from "../../dock/store.injectable";
+import callForAllLogsInjectable from "../call-for-all-logs.injectable";
+import createPodLogsTabInjectable from "../create-pod-logs-tab.injectable";
+import getRandomIdForPodLogsTabInjectable from "../get-random-id-for-pod-logs-tab.injectable";
+import type { LogTabViewModelDependencies } from "../logs-view-model";
+import { LogTabViewModel } from "../logs-view-model";
+import logsViewModelInjectable from "../logs-view-model.injectable";
+import { dockerPod } from "./pod.mock";
+
+function mockLogTabViewModel(tabId: TabId, deps: Partial<LogTabViewModelDependencies>): LogTabViewModel {
+  return new LogTabViewModel(tabId, {
+    getLogs: jest.fn(),
+    getLogsWithoutTimestamps: jest.fn(),
+    getTimestampSplitLogs: jest.fn(),
+    getLogTabData: jest.fn(),
+    setLogTabData: jest.fn(),
+    loadLogs: jest.fn(),
+    reloadLogs: jest.fn(),
+    renameTab: jest.fn(),
+    stopLoadingLogs: jest.fn(),
+    getPodById: jest.fn(),
+    getPodsByOwnerId: jest.fn(),
+    areLogsPresent: jest.fn(),
+    searchStore: new SearchStore(),
+    ...deps,
+  });
+}
+
+const getOnePodViewModel = (tabId: TabId, deps: Partial<LogTabViewModelDependencies> = {}): LogTabViewModel => {
+  const selectedPod = dockerPod;
+
+  const logs = new Map([["timestamp", "some-logs"]]);
+
+  return mockLogTabViewModel(tabId, {
+    getLogTabData: () => ({
+      selectedPodId: selectedPod.getId(),
+      selectedContainer: selectedPod.getContainers()[0].name,
+      namespace: "sadsadaasdad",
+      showPrevious: false,
+      showTimestamps: false,
+    }),
+    getPodById: (id) => {
+      if (id === selectedPod.getId()) {
+        return selectedPod;
+      }
+
+      return undefined;
+    },
+    getTimestampSplitLogs: () => [...logs],
+    getLogs: () => ["some-logs"],
+    getLogsWithoutTimestamps: () => ["some-logs"],
+    ...deps,
+  });
+};
+
+describe("download logs options in pod logs dock tab", () => {
+  let builder: ApplicationBuilder;
+  let openSaveFileDialogMock: jest.MockedFunction<() => void>;
+
+  beforeEach(() => {
+    builder = getApplicationBuilder();
+
+    builder.setEnvironmentToClusterFrame();
+
+    builder.beforeApplicationStart(({ rendererDi }) => {
+      rendererDi.override(callForAllLogsInjectable, () => () => Promise.resolve("all-logs"));
```

Comment:
I would prefer if you used `asyncFn` instead of `Promise.resolve()` because now you cannot test what happens when call has made but not resolved yet. Of what happens if call rejects.

Reply:
`asyncFn` didn't work for me. Used native jest `mockResolvedValue()` instead.